### ISSUE TITLE
harfbuzz build failed

### ISSUE THREAD

**Issue by Stefan-Schmidt** [vtorri/ewpi#1]
When trying to build all packages it fails at harfbuzz for me:

harfbuzz: installed (ret=512)
error while installing harfbuzz. see config.log or make.log

The make.log gives more clues:

```
In file included from /usr/x86_64-w64-mingw32/sys-root/mingw/include/stddef.h:7:0,
                 from /usr/lib/gcc/x86_64-w64-mingw32/7.3.0/include/stddef.h:1,
                 from /usr/include/stdlib.h:31,
                 from hb-private.hh:46,
                 from hb-icu.cc:30:
/usr/x86_64-w64-mingw32/sys-root/mingw/include/crtdefs.h:75:44: error: conflicting declaration 'typedef long long unsigned int uintptr_t'
 __MINGW_EXTENSION typedef unsigned __int64 uintptr_t;
                                            ^~~~~~~~~
In file included from hb-common.h:63:0,
                 from hb-blob.h:34,
                 from hb.h:35,
                 from hb-private.hh:38,
                 from hb-icu.cc:30:
/usr/include/stdint.h:100:27: note: previous declaration as 'typedef long unsigned int uintptr_t'
 typedef unsigned long int uintptr_t;
                           ^~~~~~~~~
In file included from /usr/include/stdlib.h:279:0,
                 from hb-private.hh:46,
                 from hb-icu.cc:30:
/usr/include/sys/types.h:109:19: error: conflicting declaration 'typedef __ssize_t ssize_t'
 typedef __ssize_t ssize_t;
                   ^~~~~~~
In file included from /usr/x86_64-w64-mingw32/sys-root/mingw/include/stddef.h:7:0,
                 from /usr/lib/gcc/x86_64-w64-mingw32/7.3.0/include/stddef.h:1,
                 from /usr/include/stdlib.h:31,
                 from hb-private.hh:46,
                 from hb-icu.cc:30:
/usr/x86_64-w64-mingw32/sys-root/mingw/include/crtdefs.h:45:35: note: previous declaration as 'typedef long long int ssize_t'
 __MINGW_EXTENSION typedef __int64 ssize_t;
                                   ^~~~~~~
In file included from /usr/include/sys/types.h:130:0,
                 from /usr/include/stdlib.h:279,
                 from hb-private.hh:46,
                 from hb-icu.cc:30:
/usr/include/bits/types/time_t.h:7:18: error: conflicting declaration 'typedef __time_t time_t'
 typedef __time_t time_t;
                  ^~~~~~
In file included from /usr/x86_64-w64-mingw32/sys-root/mingw/include/stddef.h:7:0,
                 from /usr/lib/gcc/x86_64-w64-mingw32/7.3.0/include/stddef.h:1,
                 from /usr/include/stdlib.h:31,
                 from hb-private.hh:46,
                 from hb-icu.cc:30:
/usr/x86_64-w64-mingw32/sys-root/mingw/include/crtdefs.h:138:20: note: previous declaration as 'typedef __time64_t time_t'
 typedef __time64_t time_t;
                    ^~~~~~
In file included from /usr/include/sys/sysmacros.h:41:0,
                 from /usr/include/sys/types.h:205,
                 from /usr/include/stdlib.h:279,
                 from hb-private.hh:46,
                 from hb-icu.cc:30:
/usr/include/sys/sysmacros.h: In function 'unsigned int gnu_dev_major(__dev_t)':
/usr/include/bits/sysmacros.h:44:60: warning: right shift count >= width of type [-Wshift-count-overflow]
     __major |= ((__dev & (__dev_t) 0xfffff00000000000u) >> 32); \
                                                            ^
/usr/include/sys/sysmacros.h:79:1: note: in expansion of macro '__SYSMACROS_DEFINE_MAJOR'
 __SYSMACROS_DEFINE_MAJOR (__SYSMACROS_IMPL_TEMPL)
 ^~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/sys/sysmacros.h: In function '__dev_t gnu_dev_makedev(unsigned int, unsigned int)':
/usr/include/bits/sysmacros.h:68:54: warning: left shift count >= width of type [-Wshift-count-overflow]
     __dev |= (((__dev_t) (__major & 0xfffff000u)) << 32); \
                                                      ^
/usr/include/sys/sysmacros.h:81:1: note: in expansion of macro '__SYSMACROS_DEFINE_MAKEDEV'
 __SYSMACROS_DEFINE_MAKEDEV (__SYSMACROS_IMPL_TEMPL)
 ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from hb-private.hh:55:0,
                 from hb-icu.cc:30:
/usr/x86_64-w64-mingw32/sys-root/mingw/include/intrin.h: At global scope:
/usr/x86_64-w64-mingw32/sys-root/mingw/include/intrin.h:522:5: error: conflicting declaration of C function 'int _setjmp(__jmp_buf_tag*, void*)'
     __MACHINE(int __cdecl __attribute__ ((__nothrow__,__returns_twice__)) _setjmp(jmp_buf,void *))
     ^
In file included from /usr/x86_64-w64-mingw32/sys-root/mingw/include/intrin.h:38:0,
                 from hb-private.hh:55,
                 from hb-icu.cc:30:
/usr/include/setjmp.h:58:12: note: previous declaration 'int _setjmp(__jmp_buf_tag*)'
 extern int _setjmp (struct __jmp_buf_tag __env[1]) __THROWNL;
            ^~~~~~~
In file included from hb-icu.cc:30:0:
hb-private.hh:298:1: error: static assertion failed
 static_assert ((sizeof (int64_t) == 8), "");
 ^~~~~~~~~~~~~
hb-private.hh:299:1: error: static assertion failed
 static_assert ((sizeof (uint64_t) == 8), "");
 ^~~~~~~~~~~~~
In file included from /usr/x86_64-w64-mingw32/sys-root/mingw/include/windows.h:114:0,
                 from hb-mutex-private.hh:53,
                 from hb-private.hh:617,
                 from hb-icu.cc:30:
/usr/x86_64-w64-mingw32/sys-root/mingw/include/stralign.h: In function 'int ua_wcsicmp(LPCUWSTR, LPCUWSTR)':
/usr/x86_64-w64-mingw32/sys-root/mingw/include/stralign.h:155:14: error: '_wcsicmp' was not declared in this scope
       return _wcsicmp((LPCWSTR)String1,(LPCWSTR)String2);
              ^~~~~~~~
/usr/x86_64-w64-mingw32/sys-root/mingw/include/stralign.h:155:14: note: suggested alternative: 'ua_wcsicmp'
       return _wcsicmp((LPCWSTR)String1,(LPCWSTR)String2);
              ^~~~~~~~
              ua_wcsicmp
make[3]: *** [Makefile:2290: libharfbuzz_icu_la-hb-icu.lo] Error 1
make[3]: Leaving directory '/home/stefan/EFL/efl/ewpi/packages/harfbuzz/harfbuzz-1.8.4/src'
make[2]: *** [Makefile:3051: install-recursive] Error 1
make[2]: Leaving directory '/home/stefan/EFL/efl/ewpi/packages/harfbuzz/harfbuzz-1.8.4/src'
make[1]: *** [Makefile:3432: install] Error 2
make[1]: Leaving directory '/home/stefan/EFL/efl/ewpi/packages/harfbuzz/harfbuzz-1.8.4/src'
make: *** [Makefile:516: install-recursive] Error 1
```

**Comment by vtorri** [comment]
which OS do you use ? I'll try to test ewpi on that OS

**Comment by Stefan-Schmidt** [comment]
Fedora 27

**Comment by vtorri** [comment]
fixed in 83d77436b41b6145d234c17f4e6c6ba4785ad969
